manifests/vanilla/deploy-csi-snapshot-components.sh

```diff
@@ -0,0 +1,216 @@
+else
+        echo -e "❌ ERROR: Please enable the block-volume-snapshot feature to proceed"
+        exit 1
+fi
+
+while [[ $# -gt 0 ]]; do
+    case ${1} in
+        --release)
+            release="$2"
+            shift
+            ;;
+        *)
+            usage
+            ;;
+    esac
+    shift
+done
+
+
+if [ -z "${release}" ]
+then
+  release=v4.1.1
+else
+  if [ "${release}" != "v4.1.1" ] && [ "${release}" != "v4.1.0" ]
+  then
+    echo -e "❌ ERROR: Only v4.1.1 or v4.1.0 is supported"
+    exit 1
+  fi
+fi
+echo "Using release version: ${release}"
+
+# Waits for deployment to complete. $1: name of deployment, $2: namespace.
+wait_for_deployment() {
+local deployed=false
+local requiredReplicas
+local availableReplicas
+for _ in $(seq 20); do
+    requiredReplicas=$(kubectl get deployment "$1" -n "$2" -o jsonpath='{.spec.replicas}')
+    availableReplicas=$(kubectl get deployment "$1" -n "$2" -o jsonpath='{.status.availableReplicas}')
+    if [[ ${availableReplicas} == "${requiredReplicas}" ]]; then
+        deployed=true
+        break
+    fi
+    echo "waiting for $1 to complete.."
+    sleep 10
+done
+
+if [ $deployed ]
+then
+  echo -e "✅ $1 successfully deployed!"
+else
+  echo -e "❌ ERROR: Failed to deploy $1"
+  exit 1
+fi
+}
+
+# Deploy the snapshot controller.
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/"${release}"/client/config/crd/snapshot.storage.k8s.io_volumesnapshotclasses.yaml 2>/dev/null || true
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/"${release}"/client/config/crd/snapshot.storage.k8s.io_volumesnapshotcontents.yaml 2>/dev/null || true
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/"${release}"/client/config/crd/snapshot.storage.k8s.io_volumesnapshots.yaml 2>/dev/null || true
+echo  -e "✅ Deployed VolumeSnapshot CRDs"
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/"${release}"/deploy/kubernetes/snapshot-controller/rbac-snapshot-controller.yaml 2>/dev/null || true
+echo -e "✅ Created  RBACs for snapshot-controller"
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/"${release}"/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml 2>/dev/null || true
+echo -e "✅ Deployed snapshot-controller"
+
+wait_for_deployment snapshot-controller kube-system
+
+#Deploy the external snapshot validating webhook.
+service=snapshot-validation-service
+secret=snapshot-webhook-certs
+namespace=kube-system
+
+if [ ! -x "$(command -v openssl)" ]; then
+    echo "openssl not found"
+    exit 1
+fi
+
+tmpdir=$(mktemp -d)
+echo "creating certs in tmpdir ${tmpdir} "
+
+cat <<EOF >> "${tmpdir}"/server.conf
+[req]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+prompt = no
+[req_distinguished_name]
+CN = ${service}.${namespace}.svc
+[ v3_req ]
```

Comment:
extra spaces inside brackets

Reply:
https://github.com/kubernetes-csi/external-snapshotter/blob/629e8621e088fcfd59a0f4f9f9af0a9f9da983dd/deploy/kubernetes/webhook-example/create-cert.sh#L63

Taken directly from there, perhaps it expects a space.

Reply:
Ok. Never mind. Just make sure it is not a typo.